libcontainer/nsenter/nsexec.c

```diff
@@ -516,6 +517,14 @@ void nsexec(void)
 				/* This doesn't need to be global, we're in the parent. */
 				int syncfd = syncpipe[1];
 
+				/*
+				 * Father only return when both child and grandchild are
+				 * ready, so we can receive all possible error codes
+				 * generated by children.
+				 */
+				if (ready == 2)
```

Comment:
I'd personally just change the `while` loop so it reads `while (ready < 2)` with basically the same comment. That way we can completely remove `goto out`.